### ISSUE TITLE
Better automation for release notes

### ISSUE THREAD

**Issue by ettiee** [cookpad/terraform-aws-eks#183]
Following https://github.com/cookpad/terraform-aws-eks/issues/149...
Could include better functionality for the release automation. Specifically - would be nice to have some automation to generate accurate and informative release notes.
A couple of github actions giving an idea of some automation processes...
https://github.com/Decathlon/release-notes-generator-action - generates release notes from milestones
https://github.com/lob/generate-changelog/ - intended to be used with the release action we use, writes changelog content into release body. We'd need to update our commit messages to the required format for the action

**Comment by errm** [comment]
Yeah this would be great... typically I just copy pasted the PR's that ended up in the release. It's a bit of a pain.

Also you might want to think about some automation for applying a change + release notes to all the supported release branches (in the case of a major bug / security issue)

**Comment by mtpereira** [comment]
📝 This just in: https://github.blog/2021-10-04-beta-github-releases-improving-release-experience/

**Comment by takanabe** [comment]
I've tested GitHub Releases' automatic release note generations  https://github.com/takanabe/github-releases-test/releases

This might be a good start because

* We can reduce the amount of copy pastes to link all PRs for each release

A tip I can share is that the automatic release note generations uses PR title as notes. So, PR title is very important to reduce our future editing efforts. For example, we sometimes mention that releases include breaking changes. In that case, those should explicitly mention in PRs.

**Comment by takanabe** [comment]
I also tested [custom release note template](https://github.com/takanabe/github-releases-test/blob/c953458d1dfea48e7cfefe5915b9434117aed811/.github/release.yml).

![image](https://user-images.githubusercontent.com/5056295/139629388-0d15c7f6-46e3-4b60-aefc-6cd712c05ce3.png)

The template enables us to categorize PRs based on PR labels. So, if we correctly put labels for each PR, we can reduce tons of copy/paste and editing efforts for release notes 📝  

I didn't test how multiple labels affect to the release note and will test it out.

**Comment by takanabe** [comment]
Only one category is applied if [multiple labels in the categories are  matches](https://github.com/takanabe/github-releases-test/blob/c953458d1dfea48e7cfefe5915b9434117aed811/.github/release.yml#L9-L21)
![image](https://user-images.githubusercontent.com/5056295/139630147-4bd7dcad-7873-4630-a093-f00cef0eb8fc.png)

**Comment by takanabe** [comment]
Of course, labels changed before generating release notes are affected. The following example changed the `enhancement` label to the `breaking-change` label on https://github.com/takanabe/github-releases-test/pull/9


![image](https://user-images.githubusercontent.com/5056295/139630733-17ee08eb-9a2b-4df6-a308-c44709164c3e.png)

![image](https://user-images.githubusercontent.com/5056295/139630824-608630d0-b8cb-4556-a246-505b7b2e3a11.png)

**Comment by takanabe** [comment]
The order of "What's Changed" section can be arranged with categories array https://github.com/takanabe/github-releases-test/commit/fc6924abe8a6cfae7a61abdfe1e5d6d55ec2fddf

![image](https://user-images.githubusercontent.com/5056295/139631388-82ad82c1-10d8-454f-8a1b-321b6a2e3664.png)

**Comment by takanabe** [comment]
TL;DR

* We can reduce copy/paste and editing efforts with the native release note generation feature.
* If we want to automate release note generations using GitHub native feature, we can use labels to categorize PRs.
* Release managers need to check if PRs have correct labels on them before generating release notes.


IMO, it's useful and worth trying.

**Comment by takanabe** [comment]
## Expected change

* Update the release process described in [RELEASE.md](https://github.com/cookpad/terraform-aws-eks/blob/6646f5f1214c750f5fd4c3bdac867caf128b6db2/RELEASING.md#release-procedure)
* Replace [UPGRADE.md](https://github.com/cookpad/terraform-aws-eks/blob/main/UPGRADING.md) with newly created release notes.
